### PR TITLE
Issue 2963: Update docker images to use OpenJDK image

### LINKED ISSUE #2963: Update docker images to use OpenJDK image
**Problem description**
The Pravega/BK images are currently based on an obsolete image `java:8` ([announcement](https://hub.docker.com/_/java/)).   Update to use an OpenJDK image.   Consider using `openjdk:8-alpine` to minimize the size (103MB vs 624MB for `openjdk:8`) ([ref](https://hub.docker.com/_/openjdk/)).

Note that the ZK operator uses the [official ZK image](https://hub.docker.com/_/zookeeper/), which is based on `openjdk:8-jre-alpine`.   Maybe this should be the base for Pravega also.

### PR DESCRIPTION
**Pull request by fpj**
Signed-off-by: Flavio Junqueira (fpj) <flavio.junqueira@emc.com>

**Change log description**  
* Updates base docker image according to the suggestion in issue #2963 .

**Purpose of the change**  
Fixes #2963 

**What the code does**  
Changes the docker base image in two places:
* `docker/pravega/Dockerfile`
* `docker/bookkeeper/Dockerfile`

**How to verify it**  
Build and deploy docker images.

### CODE REVIEW

File: docker/bookkeeper/Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
 
-FROM java:8
+FROM openjdk:8-alpine
 MAINTAINER Arvind Kandhare [arvind.kandhare@emc.com]
 
 RUN apt-get update \
```
Comment (shrids) [inline docker/bookkeeper/Dockerfile]: apt-get is not supported on alpine.

File: docker/bookkeeper/Dockerfile [hunk 2]
```diff
@@ -8,7 +8,7 @@
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
 
-FROM java:8
+FROM openjdk:8-alpine
```
Comment (EronWright) [inline docker/bookkeeper/Dockerfile]: Could you try `openjdk:8-jre-alpine` which what ZK uses?  I just noticed that fact.   Assumedly the difference is that the the JRE image doesn't have JDK stuff like `javac`.

**Comment by fpj** [comment]
@EronWright I'd rather not make the change until we have further tested it. I understand that having a smaller image is appealing, but I don't fully understand whether there can be other side effects out of using alpine. I'd suggest that we move to newer base image as this PR is currently doing and address the move to alpine separately.

**Comment by arvindkandhare** [comment]
> This change is primarily to address all the vulnerabilities due to the stale base image, not for size

Thanks for clarifying. Will wait for system tests to pass and then approve this.

**Pull request merged**